website/www/site/content/en/documentation/ml/preprocess-data.md

```diff
@@ -0,0 +1,227 @@
+-   Only available for pipelines that use [default windows](/documentation/programming-guide/#single-global-window).
+-   Only supports one-to-one transform mapping on a single element.
+
+## Transforms {#transforms}
+
+You can use `MLTransform` to perform the following data processing transforms.
+For information about the transforms, see
+[Module:tft](https://www.tensorflow.org/tfx/transform/api_docs/python/tft) in the
+TensorFlow documentation.
+
+{{< table >}}
+| Transform name | Description |
+| ------- | ---------------|
+| ApplyBuckets | See [`tft.apply_buckets`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/apply_buckets) in the TensorFlow documentation. |
+| Bucketize | See [`tft.bucketize`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/bucketize) in the TensorFlow documentation. |
+| ComputeAndApplyVocabulary | See [`tft.compute_and_apply_vocabulary`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/compute_and_apply_vocabulary) in the TensorFlow documentation. |
+| NGrams | See [`tft.ngrams`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/ngrams) in the TensorFlow documentation. |
+| ScaleByMinMax | See [`tft.scale_by_min_max`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/scale_by_min_max) in the TensorFlow documentation. |
+| ScaleTo01 | See [`tft.scale_to_0_1`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/scale_to_0_1) in the TensorFlow documentation. |
+| ScaleToZScore | See [`tft.scale_to_z_score`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/scale_to_z_score) in the TensorFlow documentation. |
+| TFIDF | See [`tft.tfidf`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/tfidf) in the TensorFlow documentation. |:
+{{< /table >}}
+
+You can apply the transforms on either single or multiple columns passed as a
+`dict` on structured data. Keys are column names and values are lists containing
+each column's data.
+
+## I/O requirements {#io}
+
+-   Input to the `MLTransform` class must be in one of the following formats:
+    -   A `dict` of `str`
+        -   Primitive types
+        -   List of primitive types
+        -   NumPy arrays
+-   `MLTransform` outputs a Beam `Row` object with NumPy arrays.
+-   The output `PCollection` is a schema `PCollection`. The output schema
+    contains the transformed columns.
+
+## Artifacts {#artifacts}
+
+Artifacts are additional data elements created by data transformations.
+Examples of artifacts are the minimum and maximum values from a `ScaleTo01`
+transformation, or the mean and variance from a `ScaleToZScore`
+transformation.
+
+In the `MLTransform` class, the `artifact_mode` parameter determines how the
+class handles the artifacts generated or consumed by the data transformations.
+The `artifact_mode` parameter has two states: *produce* and *consume*. The state
+determines whether the `MLTransform` class creates artifacts or retrieves
+artifacts.
```

Comment:
We had a last minute pre-release change to how this is defined. Basically, instead of

`MLTransform(artifact_location='x/y/z', artifact_mode=Produce` we now write `MLTransform(write_artifact_location='x/y/z'`)`

and likewise for consume mode we write `MLTransform(read_artifact_location='x/y/z'`)`

Sorry I missed updating you on this sooner, that's my fault. The functionality is preserved, so the only difference is the parameter naming.

Reply:
We probably want to call it read/write mode instead of produce/consume mode as well to match the parameter naming